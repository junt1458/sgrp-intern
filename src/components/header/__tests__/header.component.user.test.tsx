import { act, render } from '@testing-library/react';
import Header from '../header.component';

const dummy_token = `DUMMY.${btoa(
  JSON.stringify({
    'https://hasura.io/jwt/claims': {
      'x-hasura-default-role': 'dummy-role',
      'x-hasura-user-id': '',
    },
  })
)}.DUMMY`;

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      push: jest.fn(),
    };
  },
}));

jest.mock('@auth0/auth0-react', () => ({
  Auth0Provider: ({ children }) => <div>{children}</div>,
  useAuth0: () => ({
    isLoading: false,
    user: { sub: 'foobar' },
    isAuthenticated: true,
    loginWithRedirect: jest.fn(),
    getAccessTokenSilently: jest
      .fn()
      .mockImplementation(() => Promise.resolve(dummy_token)),
  }),
}));

it('should render the component', () => {
  act(() => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
