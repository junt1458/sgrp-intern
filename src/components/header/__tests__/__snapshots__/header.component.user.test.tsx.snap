// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `<div />`;
